Add huggingface gpt2 fake tensor unit test for torch.onnx.dynamo_export

thiagocrepaldi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115380

open llama, dolly v2 and falcon are still broken regardless of `ExportedProgram`, so they were not moved from `test_fx_to_onnx.py` to `fx_to_onnx_onnxruntime.py`.

Dolly and falcon already have tracking issues, but a tracking issue was created for open llama: https://github.com/pytorch/pytorch/issues/115552

A tracking issue was created for `xfail_if_model_type_is_exportedprogram` and `xfail_if_model_type_is_not_exportedprogram` issues with unexpected success runs: https://github.com/pytorch/pytorch/issues/115747

Commit:
Add huggingface gpt2 fake tensor unit test for torch.onnx.dynamo_export

[ghstack-poisoned]

titaiwangms commented on test/onnx/test_fx_to_onnx_with_onnxruntime.py:
```diff
@@ -1327,6 +1327,48 @@ def create_model():
         )
 
+    @pytorch_test_common.xfail_if_model_type_is_not_exportedprogram(
+        "DynamoExport does not support input mutation"
```
Why in the original test, it doesn't pop this during Dynamo?

thiagocrepaldi replied:
What do you mean? Can you elaborate?

titaiwangms replied:
Yes, so the reason huggingface gpt2 is in `test_fx_to_onnx.py` is because we observed errors when we ran the exported model with ORT, but didn't see any issue with dynamo export.

https://github.com/pytorch/pytorch/blob/686a3e0bf08af50a3b83532058fc5a6667365e51/test/onnx/test_fx_to_onnx.py#L531-L556

However, this xfail reason says there is something wrong during dynamo export. I am wondering what's the difference could make the same models/weights having different error/issue.

titaiwangms reviewed (commented):
This is great! What happens to dolly, falcon, and open llama? Is there any issue moving them here?

thiagocrepaldi commented:
> This is great! What happens to dolly, falcon, and open llama? Is there any issue moving them here?

We can add more models, but adding the missing features for fake tensor has a higher priority/impact

titaiwangms commented:
> > This is great! What happens to dolly, falcon, and open llama? Is there any issue moving them here?
> 
> We can add more models, but adding the missing features for fake tensor has a higher priority/impact

The model tests are all prepared for you, and you really can done it in 10 minutes, which can give us comprehensive idea of how completed `ExportedProgram` can be compared to dynamo_export with nn.Module.

https://github.com/pytorch/pytorch/blob/686a3e0bf08af50a3b83532058fc5a6667365e51/test/onnx/test_fx_to_onnx.py#L558-L632

Commit:
Update on "Add huggingface gpt2 fake tensor unit test for torch.onnx.dynamo_export"


tracking issues:
* https://github.com/pytorch/pytorch/issues/115552 (open llama)


[ghstack-poisoned]

thiagocrepaldi commented:
> > > This is great! What happens to dolly, falcon, and open llama? Is there any issue moving them here?
> > 
> > 
> > We can add more models, but adding the missing features for fake tensor has a higher priority/impact
> 
> The model tests are all prepared for you, and you really can done it in 10 minutes, which can give us comprehensive idea of how completed `ExportedProgram` can be compared to dynamo_export with nn.Module.
> 
> https://github.com/pytorch/pytorch/blob/686a3e0bf08af50a3b83532058fc5a6667365e51/test/onnx/test_fx_to_onnx.py#L558-L632

* dolly and falcon have bugs that were not fixed by ExportedProgram path;
* open llama is broken on ExportedProgram but I have filed https://github.com/pytorch/pytorch/issues/115552 to track it.

Commit:
Update on "Add huggingface gpt2 fake tensor unit test for torch.onnx.dynamo_export"


tracking issues:
* https://github.com/pytorch/pytorch/issues/115552 (open llama)


[ghstack-poisoned]

Commit:
Update on "Add huggingface gpt2 fake tensor unit test for torch.onnx.dynamo_export"


open llama, dolly v2 and falcon are still broken regardless of ExportedProgram, so they were not moved from test_fx_to_onnx.py to fx_to_onnx_onnxruntime.py yet

A new tracking issue was created for open llama: https://github.com/pytorch/pytorch/issues/115552


[ghstack-poisoned]

titaiwangms reviewed (approved):
Thanks!

thiagocrepaldi commented:
@pytorchbot merge

thiagocrepaldi commented:
@malfet @kit1980 Happy New Year!

Do you guys know what happened to this PR? It had the `merging` added and later replaced by `merged`, but the PR is still open and not merged? There are some referals in the timeline too that I don't get it

malfet commented:
@thiagocrepaldi Happy New Year! Sometimes GitHub fails to auto close PR, so if you see the change in think, feel free to manually close
